src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -56,6 +58,9 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             {
                 return await ProcessRequestAsync(req, function, ct);
             };
+
+            IFuncExecutor proxyFunctionExecutor = new ProxyFunctionExecutor(this._scriptHostManager, controllerContext.Configuration.DependencyResolver, this);
+            request.Properties.Add(ScriptConstants.AzureProxyFunctionExecutorKey, proxyFunctionExecutor);
             return await _scriptHostManager.HttpRequestManager.ProcessRequestAsync(request, processRequestHandler, cancellationToken);
```

Comment:
can we decide at this point whether we want to `ProcessRequestAsync` with the scriptHostManager or the proxyClient? I don't understand why attaching an executor to request properties is necessary.

Reply:
This may need more explanation. The change is trying to optimize the scenario where the backend of the proxy is another function. In such cases we want to avoid forwarding the request to the backend via frontend etc. Instead we pass a wrapper on FunctionsController (here ProxyFunctionExecutor) which will be used by the ProxyClient to reroute the request after it performs any request/response overrides. 

---
In reply to: [135931711](https://github.com/Azure/azure-webjobs-sdk-script/pull/1839#discussion_r135931711) [](ancestors = 135931711)